### PR TITLE
doc: Fix RPC result documentation

### PR DESCRIPTION
**Pull request by maflcko**
Fix:
* Incorrectly named fields
* Add missing ones
* Add missing optional flag

### CODE REVIEW

**Review by Herrytheeagle — APPROVED** [review]
Properly fixed with the comment integration for clear and understandable codebase

**Comment by practicalswift** [comment]
Concept ACK

**Review by fanquake — COMMENTED** [review]
ACK fa1d3ddaa35424ce92a1d3c31aa7a94787811281 - Clearly this is a big improvement, some of the docs here are flat out wrong.

File: src/rpc/net.cpp [hunk 1]
```diff
@@ -135,9 +135,9 @@ static RPCHelpMan getpeerinfo()
...
                             {RPCResult::Type::NUM, "timeoffset", "The time offset in seconds"},
-                            {RPCResult::Type::NUM, "pingtime", "ping time (if available)"},
-                            {RPCResult::Type::NUM, "minping", "minimum observed ping time (if any at all)"},
-                            {RPCResult::Type::NUM, "pingwait", "ping wait (if non-zero)"},
+                            {RPCResult::Type::NUM, "pingtime", /* optional */ true, "ping time (if available)"},
+                            {RPCResult::Type::NUM, "minping", /* optional */ true, "minimum observed ping time (if any at all)"},
+                            {RPCResult::Type::NUM, "pingwait", /* optional */ true, "ping wait (if non-zero)"},
                             {RPCResult::Type::NUM, "version", "The peer version, such as 70001"},
                             {RPCResult::Type::STR, "subver", "The string version"},
                             {RPCResult::Type::BOOL, "inbound", "Inbound (true) or Outbound (false)"},
```
[the remark this answers is not in the document]
Comment (MarcoFalke) [reply src/rpc/net.cpp:143]: Strictly they are conditional. Though I think the RPC should be changed to not be racy, unless there is a strong reason for it to prefer to be racy.

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -3426,6 +3429,10 @@ RPCHelpMan signrawtransactionwithwallet()
                             {
                                 {RPCResult::Type::STR_HEX, "txid", "The hash of the referenced, previous transaction"},
                                 {RPCResult::Type::NUM, "vout", "The index of the output to spent and used as input"},
+                                {RPCResult::Type::ARR, "witness", "",
```
[the remark this answers is not in the document]
Comment (MarcoFalke) [reply src/wallet/rpcwallet.cpp:3448]: It already is?

File: src/rpc/rawtransaction.cpp [hunk 3]
```diff
@@ -1078,22 +1083,23 @@ static RPCHelpMan decodepsbt()
                                 }},
```
[the remark this answers is not in the document]
Comment (MarcoFalke) [reply src/rpc/rawtransaction.cpp:1083]: Thanks, fixed.

**Review by maflcko — COMMENTED** [review]
Force pushed to reply to feedback

**Comment by MarcoFalke** [comment]
Failing CI can be ignored

**Review by fanquake — APPROVED** [review]
ACK fa10fbc6658fb8d87118cd550e4de406dca45f23

**Pull request merged**